### PR TITLE
Feature/security ci

### LINKED ISSUE #256: Security CI
I was looking at the CI of other projects and realized that we can probably just use the same configs.

1. [Slither CI - Origin](https://github.com/OriginProtocol/origin-dollar/blob/master/.github/workflows/slither.yml)
2. [Echidna CI - Uniswap](https://github.com/Uniswap/v3-core/blob/main/.github/workflows/fuzz-testing.yml)
3. [MythX CI - Uniswap](https://github.com/Uniswap/v3-core/blob/main/.github/workflows/mythx.yml)
4. [Snyk CI - Origin](https://github.com/OriginProtocol/origin-dollar/blob/master/.github/workflows/snyk.yml)

_Originally posted by @pavlovcik in https://github.com/ubiquity/ubiquity-dollar/discussions/197#discussioncomment-3606636_

### PR DESCRIPTION
**Pull request by 0xautohub**
Implemented Slither
- needed to move openzeppelin, uniswap and solidity-linked-list dependencies from root directory to node_modules directories in child projects
- Originally tried to run "yarn slither:all" in slither.yml, but it generated "not found openzeppelin file" error, so used this way

Implemented Echdina
- will need to add test sol files for Echdina test

### CODE REVIEW

File: package.json
Comment (cuddle-dev) [inline package.json]: Moving node_modules isn't a preferred way to make slither working because foundry conversion is in progress right now. It doesn't have node_modules at all. And then this is out of concept of workspace. we need to find a way without moving/copying openzeppelin.

**Review by cuddle-dev — CHANGES_REQUESTED** [review]

File: package.json
Comment (0x4007) [reply package.json]: Worst case scenario I would recommend a symlink `ln -s`

Comment (0xautohub) [reply package.json]: Okay, let me try symlink. Thanks for your directions

**Comment by 0xautohub** [comment]
Hey @pavlovcik @0xcodercrane I have just tested with symlink, but keep failing. I need your checking
image

Build link (https://github.com/ubiquity/ubiquity-dollar/actions/runs/3103846914/jobs/5027629849#step:5:259)

image
Link(https://github.com/sunny0714/ubiquity-dollar/actions/runs/3104574625/jobs/5029204912#step:5:353)

**Comment by 0xautohub** [comment]
Did you check this over? @pavlovcik

**Comment by 0x4007** [comment]
> Did you check this over? @pavlovcik

I see that you're keen on solving the other issue while you're blocked on this issue, which I think makes sense. Let me ask @sergfeldman to decide on this. In the meantime, lets wait for @0xcodercrane's response to unblock you. Thanks!

**Comment by cuddle-dev** [comment]
I just requested the changes to make it extensible but the screenshot says that it throws error in slither part.  I am thinking there could be a way to configure symlink in the config but never used it before. 

For lint checking, do you @sunny0714  want us to update lint config to resolve your issue? If yes, that would be helpful to let us know which part needs to be updated. Another concern about this pull request is that we can merge it into development branch once all the checks pass.

**Comment by 0x4007** [comment]
> I just requested the changes to make it extensible but the screenshot says that it throws error in slither part.  I am thinking there could be a way to configure symlink in the config but never used it before. 


Notice that the error code isn't 1. Meaning it's intentional that Slither is signaling to the OS that it found a security error. The program didn't crash.

**Comment by 0xautohub** [comment]
>

**Comment by 0xautohub** [comment]
> 

Commit (https://github.com/ubiquity/ubiquity-dollar/pull/268/commits/959e18ecfee4772cbd6b79918b1b95a89fbfb114)
Hope you to check the symlink in package.json of individual project and let me know if it's correct or not

**Comment by 0xautohub** [comment]
> 

I have checked the link of node_modules directories on my own, seems to work. but importing openzeppline keep failing
This is only one error now

**Comment by 0xautohub** [comment]
> I just requested the changes to make it extensible but the screenshot says that it throws error in slither part. I am thinking there could be a way to configure symlink in the config but never used it before.
> 
> For lint checking, do you @sunny0714 want us to update lint config to resolve your issue? If yes, that would be helpful to let us know which part needs to be updated. Another concern about this pull request is that we can merge it into development branch once all the checks pass.

I think this PR can be merged after all the errors generated from Slither have been fixed. Yeah, absolutely agree with @pavlovcik

**Pull request closed without merging**
